include/boost/math/statistics/univariate_statistics.hpp

```diff
@@ -231,10 +230,9 @@ inline auto first_four_moments(ExecutionPolicy&& exec, ForwardIterator first, Fo
         }
         else
         {
-            static_assert(!std::is_same_v<Real, long double>, "Error for parallel calculation using long double exceeds 100 Epsilon");
+            static_assert(!std::is_same_v<Real, long double> ^ !std::is_trivially_constructible_v<Real>, 
+                          "Error for parallel calculation using long double or aribitrary precision types is excessivly large (>100 epsilon).");
```

Comment:
Do we have a good idea of why this is happening just for multiprecision types?

For instance, is it happening for mpfr types as well?

Reply:
I tried it with `cpp_bin_float_50` and `float128`, and for both types the residual error for M2 is 5.55112e-17. As for the why? I am not quite sure. That error cascaded to others like kurtosis that use `first_four_moments` where the error remained in the realm of e-17.

Reply:
When you see that, it's strongly suggestive of a stray double precision value in the calculation somewhere.  What's the precision for 80-bit long double?  One way to try and track this down is to set a breakpoint in cpp_bin_float's convert-from-double code (see assign_float in cpp_bin_float.hpp:295) and see if it get's triggered and with what values.

Reply:
@jzmaddock You are correct. `auto` was deducing the wrong type in the calculation of M2. Everything looks fine with explicit types.